### PR TITLE
Adding comments

### PR DESCRIPTION
**Pull request by AvitalGitelman**

### CODE REVIEW

File: eatin-client/src/components/accounts/profile.tsx [hunk 1]
```diff
@@ -31,10 +34,9 @@ export const Profile: FC = () => {
         variables: { category: Category.Cake, userID: userID },
     });
 
-    const { data: savedRecipes, loading: savedRecipesLoading } = useGetRecipesConnectionIsSaved(
-        userID,
-        true,
-    );
+    // const { data: savedRecipes, loading: savedRecipesLoading } = useGetSavedRecipesQuery({
```
Comment (kostmika) [inline eatin-client/src/components/accounts/profile.tsx:37]: why do you leave it in comment? why don't you delete it?

File: eatin-client/src/components/accounts/profile.tsx [hunk 2]
```diff
@@ -81,11 +90,8 @@ export const Profile: FC = () => {
                             </AsyncDataLoaderWrapper>
                         </TabPanel>
                         <TabPanel value={1} sx={{ p: 2 }}>
-                            <AsyncDataLoaderWrapper
-                                loading={savedRecipesLoading}
-                                text="loading saved recipes..."
-                            >
-                                <RecipesCatalog recipes={savedRecipes} />
+                            <AsyncDataLoaderWrapper loading={false} text="loading saved recipes...">
```
Comment (kostmika) [inline eatin-client/src/components/accounts/profile.tsx:93]: why do you need the AsyncDataLoader if it is always false?

File: eatin-client/src/components/functions/useInsertNewUserRecipe.ts [hunk 3]
```diff
@@ -1,29 +1,62 @@
...
+    useGetUserrecipesByRecipeAndUserQuery,
+} from "../../generated/graphql";
 import { useGetUsersName } from "../hooks/useGetUsersName";
 
-export function useInsertNewUserRecipe() {
+export function useInsertNewUserRecipe(recipeIndex: number) {
     const [createSavedUserRecipe] = useCreateUserRecipesMutation();
     const [data, setData] = useState<CreateUserRecipesMutation | null | undefined>(null);
     const userID = useGetUsersName();
+    const { data: recipeConnection } = useGetUserrecipesByRecipeAndUserQuery({
```
Comment (kostmika) [inline eatin-client/src/components/functions/useInsertNewUserRecipe.ts:13]:
```suggestion
    const { data: recipeConnection } = useGetUserRecipesByRecipeAndUserQuery({
```

Comment (kostmika) [reply eatin-client/src/components/functions/useInsertNewUserRecipe.ts:13]: the CamelCase in the name is not in all the places it should

File: eatin-client/src/components/functions/useInsertNewUserRecipe.ts [hunk 4]
```diff
@@ -1,29 +1,62 @@
...
-                        is_saved: isSaved,
-                        given_comment: "",
-                        is_uploaded: false,
-                    },
-                }).then((userRecipe) => {
-                    setData(userRecipe.data);
-                });
+    const updateIsSaved = useCallback(
+        (isSaved: boolean) => {
+            const val = {
```
Comment (kostmika) [inline eatin-client/src/components/functions/useInsertNewUserRecipe.ts:19]: why do you call it val? why not a more informative name?

File: eatin-client/src/components/hooks/useGetRecipesComments.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import {
+    useGetCommentsByRecipeIndexQuery,
+    useCreateCommentsMutation,
+} from "../../generated/graphql";
+import { useGetUsersName } from "./useGetUsersName";
+import { v4 as uuidv4 } from "uuid";
+import { Comments } from "../types";
+
+export function useGetRecipesComments(recipeIndex: number) {
+    const userID = useGetUsersName();
```
Comment (kostmika) [inline eatin-client/src/components/hooks/useGetRecipesComments.ts:11]: If you get the userId, why is it called getUsersName?

File: eatin-client/src/components/hooks/useGetRecipesComments.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    const [insertNewComment] = useCreateCommentsMutation();
+    const [isLoading, setIsLoading] = useState(false);
+    const { data: comments, refetch } = useGetCommentsByRecipeIndexQuery({
+        variables: { recipeID: recipeIndex },
```
Comment (kostmika) [inline eatin-client/src/components/hooks/useGetRecipesComments.ts:15]: the recipe id is it's index? the index where? in the carousel?

File: eatin-client/src/components/hooks/useGetRecipesComments.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+            return {
+                ...comment,
+                comment_timestap: updatedTimestap(comment.comment_timestap),
+            };
+        });
+        setCurrentComments(updatedCommentes ? [...updatedCommentes] : []);
+        setIsLoading(false);
+    }, [comments]);
+
+    const updatedTimestap = (timestamp: string) => {
```
Comment (kostmika) [inline eatin-client/src/components/hooks/useGetRecipesComments.ts:31]: why is it called update? it doesn't update anything, only returns the time.

File: eatin-client/src/graphql/queries/userRecipes.query.ts [hunk 8]
```diff
@@ -84,21 +84,33 @@ export const GET_USERRECIPES_BY_USER_AND_IS_UPLOADED = gql`
...
+//         }
+//     }
+// `;
 
-export const GET_USERRECIPES_BY_USER_AND_COMMENT = gql`
-    query getUserrecipesByUserAndComment($userID: String!) {
-        userRecipesByUserAndIsCommentExists(userID: $userID) {
+export const GET_USERRECIPES_BY_RECIPE_AND_USER = gql`
+    query getUserrecipesByRecipeAndUser($userID: String!, $recipeID: Float!) {
+        userRecipesByUserAndRecipe(userID: $userID, recipeID: $recipeID) {
```
Comment (kostmika) [inline eatin-client/src/graphql/queries/userRecipes.query.ts:113]:
```suggestion
export const GET_USER_RECIPES_BY_USER_AND_RECIPE = gql`
    query getUserrecipesByUserAndRecipe($userID: String!, $recipeID: Float!) {
        userRecipesByUserAndRecipe(userID: $userID, recipeID: $recipeID) {
```

File: eatin-client/src/pages/RecipePage/comments/CommentsSection.tsx [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+        }
+    };
+
+    const insertNewCommentToSection = async () => {
+        addNewComment(newCommentVal);
+        setNewCommentVal("");
+        notify("A new comment was added!");
+    };
+
+    const timeAgo = (date: Date) => {
```
Comment (kostmika) [inline eatin-client/src/pages/RecipePage/comments/CommentsSection.tsx:40]: I think the name is not totally understood. maybe rename it/

File: eatin-client/src/components/functions/useInsertNewUserRecipe.ts (hunk 3 above)
Comment (AvitalGitelman) [reply eatin-client/src/components/functions/useInsertNewUserRecipe.ts:13]: Its generated like that by graphql

File: eatin-client/src/components/hooks/useGetRecipesComments.ts (hunk 6 above)
Comment (AvitalGitelman) [reply eatin-client/src/components/hooks/useGetRecipesComments.ts:15]: Its the ID of the recipe in the DB

File: eatin-client/src/components/hooks/useGetRecipesComments.ts (hunk 5 above)
Comment (AvitalGitelman) [reply eatin-client/src/components/hooks/useGetRecipesComments.ts:11]: Welp dunno

Comment (kostmika) [reply eatin-client/src/components/hooks/useGetRecipesComments.ts:11]: so maybe change it...?

File: eatin-client/src/components/hooks/useGetRecipesComments.ts (hunk 6 above)
Comment (kostmika) [reply eatin-client/src/components/hooks/useGetRecipesComments.ts:15]: I know the ID is for the db, I asked what is the index, index where?

**Pull request merged**